dataflow/flex-templates/streaming_beam/README.md

```diff
@@ -0,0 +1,282 @@
+[JSON encoded](https://www.w3schools.com/whatis/whatis_json.asp) messages from
+[Pub/Sub](https://cloud.google.com/pubsub),
+transforms the message data, and writes the results to a
+[BigQuery](https://cloud.google.com/bigquery) table.
+
+* [Dockerfile](Dockerfile)
+* [streaming_beam.py](streaming_beam)
+* [metadata.json](metadata.json)
+
+### Building a container image
+
+We will build the
+[Docker](https://docs.docker.com/engine/docker-overview/)
+image for the Apache Beam pipeline.
+We are using
+[Cloud Build](https://cloud.google.com/cloud-build)
+so we don't need a local installation of Docker.
+
+> *Note:* You can speed up subsequent builds with
+> [Kaniko cache](https://cloud.google.com/cloud-build/docs/kaniko-cache)
+> in Cloud Build.
+>
+> ```sh
+> # (Optional) Enable to use Kaniko cache by default.
+> gcloud config set builds/use_kaniko True
+> ```
+
+Cloud Build allows you to
+[build a Docker image using a `Dockerfile`](https://cloud.google.com/cloud-build/docs/quickstart-docker#build_using_dockerfile).
+and saves it into
+[Container Registry](https://cloud.google.com/container-registry/),
+where the image is accessible to other Google Cloud products.
+
+```sh
+export TEMPLATE_IMAGE="gcr.io/$PROJECT/samples/dataflow/streaming-beam:latest"
+
+# Build the image into Container Registry, this is roughly equivalent to:
+#   gcloud auth configure-docker
+#   docker image build -t $TEMPLATE_IMAGE .
+#   docker push $TEMPLATE_IMAGE
+gcloud builds submit --tag "$TEMPLATE_IMAGE" .
+```
+
+Images starting with `gcr.io/PROJECT/` are saved into your project's
+Container Registry, where the image is accessible to other Google Cloud products.
+
+### Creating a Flex Template
+
+To run a template, you need to create a *template spec* file containing all the
+necessary information to run the job, such as the SDK information and metadata.
+
+The [`metadata.json`](metadata.json) file contains additional information for
+the template such as the "name", "description", and input "parameters" field.
+
+The template file must be created in a Cloud Storage location,
+and is used to run a new Dataflow job.
+
+```sh
+export TEMPLATE_PATH="gs://$BUCKET/samples/dataflow/templates/streaming-beam.json"
+
+# Build the Flex Template.
+gcloud beta dataflow flex-template build $TEMPLATE_PATH \
+  --image "$TEMPLATE_IMAGE" \
+  --sdk-language "PYTHON" \
+  --metadata-file "metadata.json"
+```
+
+The template is now available through the template file in the Cloud Storage
+location that you specified.
+
+### Running a Dataflow Flex Template pipeline
+
+You can now run the Apache Beam pipeline in Dataflow by referring to the
+template file and passing the template
+[parameters](https://cloud.devsite.corp.google.com/dataflow/docs/guides/specifying-exec-params#setting-other-cloud-dataflow-pipeline-options)
+required by the pipeline.
+
+```sh
+# Parameters before the `--` are passed to the `gcloud` command.
```

Comment:
These comments are now outdated, please change to:

```
# Run the Flex Template.
```

Reply:
Done.